### PR TITLE
Add command completion and history for TurtleCoind

### PR DESCRIPTION
**Pull request by rixombea**

### CODE REVIEW

**Comment by brandonlehmann** [comment]
@rixombea can you squash this down into a reasonable number of commits please?

**Comment by brandonlehmann** [comment]
Also, you may want to bring your development branch up to date with the current development branch before tossing these changes in. That will clean up the commit history here quite a bit.

**Comment by brandonlehmann** [comment]
@rixombea thanks for cleaning that up :) will take a look :)

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -92,6 +92,11 @@ target_link_libraries(WalletService Mnemonics)
 if(MSVC)
 	target_link_libraries(TurtleCoind P2P Rpc Serialization System Http Logging CryptoNoteCore Crypto Common rocksdb ${Boost_LIBRARIES} )
 else()
+    if(ENABLE_LINENOISE)
+        target_link_libraries(TurtleCoind P2P Rpc Serialization System Http Logging CryptoNoteCore Crypto Common rocksdblib linenoise ${Boost_LIBRARIES} )
+    else()
+        target_link_libraries(TurtleCoind P2P Rpc Serialization System Http Logging CryptoNoteCore Crypto Common rocksdblib ${Boost_LIBRARIES} )
+    endif()
 	target_link_libraries(TurtleCoind P2P Rpc Serialization System Http Logging CryptoNoteCore Crypto Common rocksdblib ${Boost_LIBRARIES} )
```
[the remark this answers is not in the document]
Comment (brandonlehmann) [reply src/CMakeLists.txt]: Yes

**Comment by RocksteadyTC** [comment]
Still simmering?

**Comment by brandonlehmann** [comment]
Yes. There's a dupe line in there

**Comment by RocksteadyTC** [comment]
cool

File: src/CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (rixombea) [reply src/CMakeLists.txt]: Yes &  fixed

File: src/Common/ConsoleHandler.cpp [hunk 2]
```diff
@@ -240,6 +263,35 @@ namespace Common {
...
+                linenoiseFree(cline);
+                boost::algorithm::trim(line);
+                std::cout << "" << std::endl;
+                if (!line.empty()) {
+                    handleCommand(line);
+                }
+            }else{
+                break;
+            }
+            //std::cout << m_prompt;
```
[the remark this answers is not in the document]
Comment (rixombea) [reply src/Common/ConsoleHandler.cpp]: done.

**Pull request merged**